checks.d/etcd.py

```diff
@@ -133,18 +146,20 @@ def check(self, instance):
             self.service_check(self.SERVICE_CHECK_NAME, AgentCheck.OK,
                                tags=["url:{0}".format(url)])
 
-    def _get_self_metrics(self, url, timeout):
-        return self._get_json(url + "/v2/stats/self", timeout)
+    def _get_self_metrics(self, url, ssl_params, timeout):
+        return self._get_json(url + "/v2/stats/self",  ssl_params, timeout)
 
-    def _get_store_metrics(self, url, timeout):
-        return self._get_json(url + "/v2/stats/store", timeout)
+    def _get_store_metrics(self, url, ssl_params, timeout):
+        return self._get_json(url + "/v2/stats/store",  ssl_params, timeout)
 
-    def _get_leader_metrics(self, url, timeout):
-        return self._get_json(url + "/v2/stats/leader", timeout)
+    def _get_leader_metrics(self, url, ssl_params, timeout):
+        return self._get_json(url + "/v2/stats/leader", ssl_params, timeout)
 
-    def _get_json(self, url, timeout):
+    def _get_json(self, url, ssl_params, timeout):
         try:
-            r = requests.get(url, timeout=timeout, headers=headers(self.agentConfig))
+            certificate = (ssl_params['ssl_certfile'], ssl_params['ssl_keyfile']) if 'ssl_certfile' in ssl_params and 'ssl_keyfile' in ssl_params else None
```

Comment:
nitpick but you should use ternary conditions only when it makes the code easier to read.

In that case as it makes a pretty long line i would simply go with:
```python
certificate = None
if  'ssl_certfile' in ssl_params and 'ssl_keyfile' in ssl_params:
    certificate = (ssl_params['ssl_certfile'], ssl_params['ssl_keyfile'])
```